m3u8/writer.go

```diff
@@ -1076,3 +1199,50 @@ func (p *MediaPlaylist) GetAllSegments() []*MediaSegment {
 /*
 [Protocol Version Compatibility]: https://datatracker.ietf.org/doc/html/draft-pantos-hls-rfc8216bis-16#section-8
 */
+
+/// Helper functions
+
+func splitUriBy(uri, sep string) (string, string) {
+	// split the uri by the last dot
+	uriParts := strings.Split(uri, sep)
+	if len(uriParts) < 2 {
+		return "", ""
+	}
+	// get the last part of the uri
+	lastPart := uriParts[len(uriParts)-1]
+	// get the rest of the uri
+	rest := strings.Join(uriParts[:len(uriParts)-1], sep)
+	return rest, lastPart
+}
+
+// find the numbers in the string
+// e.g., fileSequence250 -> 250
+// filePart250 -> 250
+func getSequenceNum(uri string) uint64 {
+	// find the last number in the uri
+	re := regexp.MustCompile(`(\d+)$`)
```

Comment:
Don't compile the regular expression in every function call. Do it once and keep it as a package global variable.